third_party/proxy/service.go

```diff
@@ -132,72 +135,98 @@ func (info *BaseServiceInfo) LoadBalancerIPStrings() []string {
 	return ips
 }
 
-// OnlyNodeLocalEndpoints is part of ServicePort interface.
-func (info *BaseServiceInfo) OnlyNodeLocalEndpoints() bool {
-	return info.onlyNodeLocalEndpoints
+// NodeLocalExternal is part of ServicePort interface.
+func (info *BaseServiceInfo) NodeLocalExternal() bool {
+	return info.nodeLocalExternal
+}
+
+// NodeLocalInternal is part of ServicePort interface
+func (info *BaseServiceInfo) NodeLocalInternal() bool {
+	return info.nodeLocalInternal
+}
+
+// InternalTrafficPolicy is part of ServicePort interface
+func (info *BaseServiceInfo) InternalTrafficPolicy() *v1.ServiceInternalTrafficPolicyType {
+	return info.internalTrafficPolicy
 }
 
 // TopologyKeys is part of ServicePort interface.
 func (info *BaseServiceInfo) TopologyKeys() []string {
 	return info.topologyKeys
 }
 
+// HintsAnnotation is part of ServicePort interface.
+func (info *BaseServiceInfo) HintsAnnotation() string {
+	return info.hintsAnnotation
+}
+
 func (sct *ServiceChangeTracker) newBaseServiceInfo(port *v1.ServicePort, service *v1.Service) *BaseServiceInfo {
-	onlyNodeLocalEndpoints := false
+	nodeLocalExternal := false
 	if utilproxy.RequestsOnlyLocalTraffic(service) {
-		onlyNodeLocalEndpoints = true
+		nodeLocalExternal = true
 	}
+	nodeLocalInternal := utilproxy.RequestsOnlyLocalTrafficForInternal(service)
```

Comment:
just to confirm, if `nodeLocalInternal` is `true` (if `service.Spec.InternalTrafficPolicy` is set to `ServiceInternalTrafficPolicyLocal`), it will have no effect as it will be ignored by AntreaProxy?

Reply:
No. There is no reference for `NodeLocalInternal` func in AntreaProxy.

Reply:
No as in "there will be no effect", or no as in "the statement is wrong"?

Reply:
There will be no effect for nodeLocalInternal.